### PR TITLE
Refactor additional properties

### PR DESCRIPTION
**Pull request by mbj**
### Acceptance

* [x] Verify non dirty worktree after generation + building:

```
stack build --fast gen --exec "bash -c \"stack exec -- gen generate models && stack test stripe-client-models --fast\""
```

### CODE REVIEW

File: gen/src/Gen/Schema.hs [hunk 1]
```diff
@@ -202,6 +204,11 @@ visitObjectType context schemaObject = do
         withContext :: Schema.PropertyName -> Schema.Schema -> m State
         withContext = visitSchema . pushContextProperty context
 
+    noAdditionalProperties :: Maybe Schema.AdditionalProperties -> Bool
+    noAdditionalProperties = \case
```
Comment (mbj) [inline gen/src/Gen/Schema.hs]: In the upcoming API client generation both forms are used.

At the openapi parser we intentionally do not normalize between them as they have different (yet same semantics) representations at the openapi JSON document and we do not wish to loose this information.

**Comment by mbj** [comment]
@dkubb, @snusnu This is ready for review.

**Comment by snusnu** [comment]
@mbj I think that the 2nd commit message should refer to `Pretty` (starting with a capital letter). I'm thinking that you're referring to the `Gen.Pretty` module as commit subject, and that just like in ruby, we're "dropping" the toplevel namespace (`Gen`).

File: gen/src/Gen/CLI.hs [hunk 2]
```diff
@@ -18,14 +18,27 @@ run
...
+    generateCommands :: Options.Parser (IO ())
+    generateCommands
+      = Options.subparser
+      $  mkCommand "all"    (generate allPackages)     "Generate all packages"
+      <> mkCommand "models" (generate [modelsPackage]) "Generate models package"
+
     mkCommand
       :: String
-      -> IO ()
+      -> a
```
Comment (snusnu) [inline gen/src/Gen/CLI.hs]: @mbj Would it be possible to generalize that type in a precommit already? If so, would we even consider this to be an improvement?

Comment (mbj) [reply gen/src/Gen/CLI.hs]: feel free to do this transform if it compiles individually.

File: gen/src/Gen/CLI.hs [hunk 3]
```diff
@@ -18,14 +18,27 @@ run
...
+      = Options.command "generate"
+      $ Options.info
+        (Options.helper <*> generateCommands)
+        (Options.progDesc "generator")
+
+    generateCommands :: Options.Parser (IO ())
+    generateCommands
+      = Options.subparser
+      $  mkCommand "all"    (generate allPackages)     "Generate all packages"
+      <> mkCommand "models" (generate [modelsPackage]) "Generate models package"
```
Comment (mbj) [inline gen/src/Gen/CLI.hs]: With the added `api` (future PR) we'll get another target here.

**Pull request merged**